packages/airnode-protocol/contracts/psp/BeaconServer.sol

```diff
@@ -0,0 +1,435 @@
+            msg.sender,
+            requestId,
+            sponsorWallet
+        );
+    }
+
+    /// @notice Called by AirnodeRrp to fulfill the request
+    /// @dev It is assumed that the fulfillment will be made with a single
+    /// point of data of type `int256`
+    /// @param requestId ID of the request being fulfilled
+    /// @param data Fulfillment data (a single `int256` encoded as `bytes`)
+    function fulfill(bytes32 requestId, bytes calldata data)
+        external
+        override
+        onlyAirnodeRrp
+    {
+        bytes32 templateId = requestIdToTemplateId[requestId];
+        require(templateId != bytes32(0), "No such request made");
+        delete requestIdToTemplateId[requestId];
+        (int256 decodedData, uint256 timestamp) = abi.decode(
+            data,
+            (int256, uint256)
+        );
+        require(
+            decodedData >= type(int224).min && decodedData <= type(int224).max,
+            "Value typecasting error"
+        );
+        require(timestamp <= type(uint32).max, "Timestamp typecasting error");
+        require(block.timestamp - 1 hours < timestamp, "Fulfillment stale");
+        require(
+            templateIdToBeacon[templateId].timestamp < timestamp,
+            "Fulfillment older than beacon"
+        );
+        templateIdToBeacon[templateId] = Beacon({
+            value: int224(decodedData),
+            timestamp: uint32(timestamp)
+        });
+        emit UpdatedBeacon(
+            templateId,
+            requestId,
+            int224(decodedData),
+            uint32(timestamp)
+        );
+    }
+
+    function fulfillPsp(bytes32 subscriptionId, bytes calldata data) external {
+        require(msg.sender == airnodePsp, "Sender not AirnodePsp");
+        (bytes32 templateId, , , , , , bytes memory parameters) = IAirnodePsp(
+            airnodePsp
+        ).subscriptions(subscriptionId);
+        require(parameters.length == 0, "Subscription has parameters");
+
+        (int256 decodedData, uint256 timestamp) = abi.decode(
+            data,
+            (int256, uint256)
+        );
+        require(
+            decodedData >= type(int224).min && decodedData <= type(int224).max,
+            "Value typecasting error"
+        );
+        require(timestamp <= type(uint32).max, "Timestamp typecasting error");
+        require(block.timestamp - 1 hours < timestamp, "Fulfillment stale");
+        require(
+            templateIdToBeacon[templateId].timestamp < timestamp,
+            "Fulfillment older than beacon"
+        );
+        templateIdToBeacon[templateId] = Beacon({
+            value: int224(decodedData),
+            timestamp: uint32(timestamp)
+        });
+        /*emit UpdatedBeacon(
+            templateId,
+            requestId,
+            int224(decodedData),
+            uint32(timestamp)
+        );*/
+    }
+
+    function setUpdatePercentageThreshold(
+        bytes32 subscriptionId,
+        uint256 updatePercentageThreshold
```

Comment:
How is this percentage represented? Something like 5% is equal to 5 * 10 ** 18?

Reply:
10**18 means 100%, everything else relates linearly

Reply:
https://github.com/api3dao/airnode/pull/666#discussion_r746364027